### PR TITLE
Remove reliance on GitSCMSource

### PR DESCRIPTION
**Pull request by atl-dyon**
Removes reliance on the GitSCM for BitbucketSCMSource. 

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by atl-dyon** [comment]
I attempted to convert the GitSCM to BitbucketSCM, but i feel like this should be a seperate PR as it requires a bit more effort than a simple conversion. So i have left it out of this PR and will be looking more into it soon.

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java [hunk 1]
```diff
@@ -288,11 +248,29 @@ public void setWebhookRegistered(boolean webhookRegistered) {
...
+    @SuppressWarnings("unused") // Used by XStream
+    protected Object readResolve() {
+        // Convert any legacy traits into their new equivalents
+        if (traits != null) {
+            traits = traits.stream().map(BitbucketLegacyTraitConverter::maybeConvert)
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList());
+        }
+
+        validateInitialized();
```
Comment (jaguilar-atl) [inline src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java:272]:
```suggestion
        // initialized will always be null as transient fields are not persisted so we need to reassign it here
        initialized = new AtomicBoolean(false);
        validateInitialized();
```

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java [hunk 2]
```diff
@@ -68,8 +68,17 @@
...
 
     private static final Logger LOGGER = Logger.getLogger(BitbucketSCMSource.class.getName());
-    private CustomGitSCMSource gitSCMSource;
+    @UpgradeHandled(handledBy = "static fields not persisted", removeAnnotationInVersion = "4.0.0")
+    private static final String REFSPEC_DEFAULT = "+refs/heads/*:refs/remotes/@{remote}/*";
+    @UpgradeHandled(handledBy = "initialize() method", removeAnnotationInVersion = "4.0.0")
+    private String cloneUrl;
+    @SuppressWarnings("unused") // Kept for backward compatibility
+    private transient CustomGitSCMSource gitSCMSource;
+    @UpgradeHandled(handledBy = "initialize() method", removeAnnotationInVersion = "4.0.0")
```
Comment (jaguilar-atl) [inline src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java:77]: nit:
```suggestion
    @UpgradeHandled(handledBy = "readResolve() method", removeAnnotationInVersion = "4.0.0")
```

Comment (dkjellin) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java]: But it is not.. read-resolve will throw an NPE when it calls `validateInitialized` 
Thus after re-start you will lose all your BitbucketSCMSource settings for the job.

**Review by dkjellin — CHANGES_REQUESTED** [review]
This is not upgrade, or even restart safe, the constructor is not run on deserialisation  so the `initialized` will be `null` for all but newly created `BitbucketSCMSources`

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java (hunk 2 above)
Comment (jaguilar-atl) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java]: That's true. I've also [mentioned this in a separate comment](https://github.com/jenkinsci/atlassian-bitbucket-server-integration-plugin/pull/418#discussion_r1283890631) which I believe @atl-dyon has already addressed in a follow-up commit.

**Comment by jaguilar-atl** [comment]
> This is not upgrade, or even restart safe, the constructor is not run on deserialisation so the `initialized` will be `null` for all but newly created `BitbucketSCMSources`

I think @atl-dyon has already addressed this in a [follow-up commit here](https://github.com/jenkinsci/atlassian-bitbucket-server-integration-plugin/pull/418/commits/931da991fed6ed768ec006d8b24e7baf77f53961).

**Review by jaguilar-atl — APPROVED** [review]

**Review by dkjellin — APPROVED** [review]

**Pull request merged**